File: Particle/src/Emitter.cpp
#include "Emitter.h"
#include <iostream>

Emitter::Emitter(const Vec3 &_pos, size_t _numParticles)
{
  m_pos=_pos;
  m_numParticles=_numParticles;
  m_particles.resize(_numParticles);
  for(auto &p : m_particles)
  {
    resetParticle(p);
  }
}

void Emitter::update(float _deltaT)
{
  std::cout<<"update "<<_deltaT<<'\n';
}
void Emitter::saveFrame(int _frameNo)
{
  std::cout<<"save "<<_frameNo<<'\n';
}

void Emitter::resetParticle(Particle &io_p)
{
  io_p.pos=m_pos;
}

File: Particle/Emitter.py
from Particle import Particle


class Emitter:
    def __init__(self, pos, num_particles):
        self._position = pos
        self._num_particles = num_particles
        self._particles = []
        self._init_particles()

    def _init_particles(self):
        for _ in range(self.num_particles):
            particle = Particle()
            self._particles.append(particle)

    @property
    def position(self):
        return self._position

    @property
    def num_particles(self):
        return self._num_particles

    @property
    def particles(self):
        return self._particles

    def render(self):
        for p in self._particles:
            print(p)
